### PR TITLE
feat: Add demo code to handle spartan link

### PR DESCRIPTION
**Pull request by bryans99**
The demo embed extension demonstrates how to parse a link that is
can be recieved by a spartan user from a schedule dialog in an embed.
The link is receieved in the link param and is subsequently mapped to
its own route format.

Additionally:
1. Reduces the amount of data when requesting dashboard data.
2. Renames dashboard-next link to dashboard (this was changed in the
Looker UI.
3. Adds a command to start demo embed dev server from the root of the
repo (more commands will be added).
4. Updates gitignore to ignore direnv stuff.

### CODE REVIEW

File: react/javascript/demo-embeds/src/DemoEmbeds.js [hunk 1]
```diff
@@ -52,6 +56,54 @@ export const DemoEmbeds = () => {
   })
   const { onSelectTab, selectedIndex } = tabs
 
+  useEffect(() => {
+    if (search) {
+      // Spartan demo code.
+      // Handle a link sent from a embed to a spartan user.
```
Comment (jkaster) [inline react/javascript/demo-embeds/src/DemoEmbeds.js:62]:
```suggestion
      // Handle a link sent from an embed to a spartan user.
```

trivial typo

**Review by jkaster — APPROVED** [review]
Trivial typo. Looks good.

**Review by jkaster — APPROVED** [review]
LGTM

File: react/javascript/demo-embeds/src/hooks/use_spartan_link.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ */
+export const useSpartanLink = () => {
+  const [linkParam, setLinkParam] = useState()
+  const { search, pathname } = useLocation()
+  const history = useHistory()
+
+  useEffect(() => {
+    if (search) {
+      const searchParams = new URLSearchParams(search)
+      const link = searchParams.get('link')
```
Comment (whscullin) [inline react/javascript/demo-embeds/src/hooks/use_spartan_link.jsx:50]: Renamed this?

Comment (bryans99) [reply react/javascript/demo-embeds/src/hooks/use_spartan_link.jsx:50]: https://github.com/looker-open-source/extension-examples/pull/63

Got distracted and thought I had pushed my changes up!

**Pull request merged**